### ISSUE TITLE
lambda declaration appears not to support throws clause

### ISSUE THREAD

**Issue by russel** [chapel-lang/chapel#16580]
The file:
```chapel
var f = lambda (n:int): int { return 1; };
```
compiles fine. However, this would not allow for exceptions to be propagated out. Whilst a `proc` definition can have a throws clause, it seems that `lambda` definitions do not support this. The files:

_lambda_throws_first.chpl_:
```chapel
var f = lambda (n:int): throws int { return 1; };
```

_lambda_throws_last.chpl_:
```chapel
var f = lambda (n:int): int throws { return 1; };
```

lead to the error message:
```
chpl -o lambda_throws_first lambda_throws_first.chpl
lambda_throws_first.chpl:1: syntax error: near 'throws'

chpl -o lambda_throws_last lambda_throws_last.chpl
lambda_throws_last.chpl:1: syntax error: near 'throws'
```

**Comment by bradcray** [comment]
Chapel developers: Would supporting this be as simple as propagating the `throws` clause to the `this` method that we define on the class representing the lambda?

**Comment by bradcray** [comment]
@dlongnecke-cray : Just stumbled across this unanswered question which might be right up your alley and something to keep in mind going forward with FCFs/closures.

**Comment by bradcray** [comment]
(Oh wait, maybe you already have added support for this?  At least, if I change `lambda` to `proc`, I make it through the compiler OK.  Although now I'm seeing that if I add a `throw` statement to my lambda, it complains about not being allowed to `throw` in a non-`throwing` function).

**Comment by dlongnecke-cray** [comment]
I can patch the lambda syntax to support throws since I'm already adjusting the parser in some of my recent work. It's not too much trouble.

**Comment by bradcray** [comment]
> I can patch the lambda syntax to support throws since I'm already adjusting the parser in some of my recent work

I don't think it's worth the effort since we'll be adding a deprecation warning for lambda soon (I hope) anyway.  Rather than wrestling with the parser, getting `throw` statement working with the new syntax would be far more valuable (though I recognize it's also likely a higher cost).

**Comment by dlongnecke-cray** [comment]
Resolved by #20788. Note that in the future we will likely be deprecating `lambda` syntax entirely.
